### DISCUSSION TITLE
Intercept Paste from plugins (e.g. convert paste of spreadsheet into markdown table)

### DISCUSSION THREAD

**Discussion by jespino** [mattermost/community#130] [Enhancements]
Allow plugins to intercept the paste event of the message input to add support for special types of pasting, like spreadsheed or docs.

**Comment by esethna** [comment]
I think we should just build this into the core product as long as it's tested thoroughly. We should finish your smart-paste feature ;)

**Reply by esethna** [reply]
Copy and paste is such a key use case, we really need to optimize it
